shell/platform/android/platform_view_android_jni_impl.cc

```diff
@@ -1289,12 +1277,34 @@ void PlatformViewAndroidJNIImpl::SurfaceTextureGetTransformMatrix(
   FML_CHECK(CheckException(env));
 
   float* m = env->GetFloatArrayElements(transformMatrix.obj(), nullptr);
-  float scaleX = m[0], scaleY = m[5];
-  const SkSize scaled = ScaleToFill(scaleX, scaleY);
+
+  // SurfaceTexture 4x4 Column Major -> Skia 3x3 Row Major
+
+  // SurfaceTexture 4x4 (Column Major):
+  // | m[0] m[4] m[ 8] m[12] |
+  // | m[1] m[5] m[ 9] m[13] |
+  // | m[2] m[6] m[10] m[14] |
+  // | m[3] m[7] m[11] m[15] |
+
+  // According to Android documentation, the 4x4 matrix returned should be used
+  // with texture coordinates in the form (s, t, 0, 1). Since the z component is
+  // always 0.0, we are free to ignore any element that multiplies with the z
+  // component. Converting this to a 3x3 matrix is easy:
+
+  // SurfaceTexture 3x3 (Column Major):
+  // | m[0] m[4] m[12] |
+  // | m[1] m[5] m[13] |
+  // | m[3] m[7] m[15] |
+
+  // Skia (Row Major):
+  // | m[0] m[1] m[2] |
+  // | m[3] m[4] m[5] |
+  // | m[6] m[7] m[8] |
```

Comment:
Why is the skia format described here?  I'm not seeing a translation of the matrix to convert between column major and row major.  The matrix3 format looks like like the column major format you describe.

Reply:
If this were a row-major -> row-major conversion, the matrix accesses below would be in the form:
```
m[0], m[1], m[2],
...
```
since this would produce an identical matrix.

The formats described above are the formats when written in standard matrix notation, which is row-major when read from left to right.